### ISSUE TITLE
1.13 FastHeal

### ISSUE THREAD

**Issue by plagoutte** [CaptainObvious0/NoCheatPlus#50]
### Full output of /ncp version
```
>ncp version
[16:11:30 INFO]: ---- Version information ----
[16:11:30 INFO]: #### Server ####
[16:11:30 INFO]: git-Spigot-f09662d-7c395d4 ~MC: 1.13.2~
[16:11:30 INFO]:   detected: 1.13.2
[16:11:30 INFO]: #### NoCheatPlus ####
[16:11:30 INFO]: Plugin: 1.3-beta.4-CustomFork
[16:11:30 INFO]: MCAccess: 1.4.5-1.12.2|? / CB-Reflect
[16:11:30 INFO]: Features:
[16:11:30 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11 | BlocksMC1_12 | BlocksMC1_13
[16:11:30 INFO]:   checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
[16:11:30 INFO]:   defaults: pvpKnockBackVelocity
[16:11:30 INFO]:   packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
[16:11:30 INFO]: Hooks: AllViolations~NCP~ 1.0
[16:11:30 INFO]: #### Related Plugins ####
[16:11:30 INFO]: ProtocolLib v4.4.0 | ViaVersion v2.0.0
>
```

### Describe the issue
FastHeal check considers a player is cheating if he/she is restoring more than 1 healthpoint per 4 seconds (by default, in the config)
But with 1.13, we can restore 1 healthpoint per 0.5 seconds if the foodbar is full and foodSaturationLevel > 0.
So, if a player eat to restore its life, FastHeal will be triggered. 

We can modify the config, but it will allow players to use FastHeal cheat at will, up to 1hp/0.5s. The best thing to do would be to check if a player is allowed to restore 1hp/0.5s when it occurs.

### How to reproduce the issue
Just hurt a player enough and ask him to eat.

### Extra links/Videos (Including debug logs)
Not useful here.

### Any possible config options changed or plugins that may cause interference?
No.

**Comment by MarkElf** [comment]
Fastheal disables on startup 1.9 and above. I'm assuming this is something asofold never got around to properly completing and just disabled the check as other things took priority.

**Comment by CaptainObvious0** [comment]
I'm not fully understanding this, could you provide a video? As MarkElf said, the check automatically disables on versions 1.9+.

**Comment by Lysandr0** [comment]
Probably related: [#544](https://github.com/NoCheatPlus/Issues/issues/544)
Had to manually disable it because it would still activate on start-up
(Side note: Are Regen cheats and the like still a thing? I thought they were patched up in 1.9... ([Asofold’s quote](https://github.com/NoCheatPlus/Issues/issues/122#issuecomment-208106083))Besides, if fast enough they should still be picked up by FlyingFrequency (if I’m not mistaken))

**Comment by plagoutte** [comment]
Thank you for your replies. I doesn't know that FastHeal was fixed in the game, so I desactivated now, but I had to do this manually. Here's the startup logs :
```
:56:41] [Server thread/INFO]: [NoCheatPlus] Enabling NoCheatPlus v1.3-beta.4-CustomFork
[10:56:41] [Server thread/WARN]: [NoCheatPlus] The Minecraft version seems to be more recent than the one Compat-CB-Reflect has been built with - this might work, but there could be incompatibilities.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] McAccess set to: 1.4.5-1.12.2|? / CB-Reflect
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.5 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.6.1 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.7.2 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.8 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.9 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.10 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.11 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.12 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Added block-info for Minecraft 1.13 blocks.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Applied block patches for multi client protocol support: water_lily, soil, grass_path
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Update block-info: Block shape patch for multi client protocol support around 1.7.x - 1.13.x.
[10:56:41] [Server thread/WARN]: [NoCheatPlus] Bad block id (compatibility.blocks.overrideflags): piston_moving_piece
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Inventory checks: FastConsume is available, disabled InstantEat.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Adding packet level hooks for ProtocolLib (MC 1.13.2)...
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Confirm teleport packet available (via name): TELEPORT_ACCEPT[class=PacketPlayInTeleportAccept, id=0]
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Available (and activated) packet level hooks: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Force disable FastHeal on Minecraft 1.9 and later.
[10:56:41] [Server thread/INFO]: [NoCheatPlus] Version 1.3-beta.4-CustomFork is enabled.
[10:56:41] [Server thread/WARN]: [NoCheatPlus] You are running a modified version of NCP. This version should only be used on 1.13 servers. This is only intended as a band-aid fix while NCP can be properly updated to 1.13. You may be sacrificing cheat protection for compatibility
```

**Comment by CaptainObvious0** [comment]
Issue should be fixed in latest commit.
